core/modules/widgets/select.js

```diff
@@ -53,6 +53,17 @@ Handle a change event
 SelectWidget.prototype.handleChangeEvent = function(event) {
 	var value = this.getSelectDomNode().value;
 	this.wiki.setText(this.selectTitle,this.selectField,this.selectIndex,value);
+
+	this.activeChild = 2*(this.getSelectDomNode().selectedIndex)+1;
```

Comment:
Well, doing it this way breaks if you build your options using a list widget, I changed it so that it doesn't break but it only triggers action widgets that are in explicitly given options, not in ones generated by lists.
It still isn't very generic, but I don't know how else to find the wikitext content of the selected option.